From 739175d198372a3226ce061b118d8fe5bbc8f6bf Mon Sep 17 00:00:00 2001
From: Chase Douglas <chase.douglas@canonical.com>
Date: Wed, 16 May 2012 07:49:25 -0700
Subject: [PATCH 4/6] Reset open slots array on device disable

The open slots array is used for clickpad cumulative delta computation.
If the array is not reset and becomes corrupted during the device
disable/enable cycle, the cumulative deltas may be wrong. This manifests
as jumpy cursor behavior on some clickpads after suspend/resume.

Signed-off-by: Chase Douglas <chase.douglas@canonical.com>
Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
(cherry picked from commit 0054b144f3daf00e46a35b2f165befb209df94fc)
---
 src/synaptics.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/synaptics.c b/src/synaptics.c
index 074bcb8..78660f4 100644
--- a/src/synaptics.c
+++ b/src/synaptics.c
@@ -1153,6 +1153,9 @@ SynapticsReset(SynapticsPrivate * priv)
     priv->lastButtons = 0;
     priv->prev_z = 0;
     priv->prevFingers = 0;
+#ifdef HAVE_MULTITOUCH
+    memset(priv->open_slots, 0, priv->num_slots * sizeof(int));
+#endif
 }
 
 static Bool
-- 
1.7.9.5

